Sources/ApolloCodegenLib/Frontend/ValidationOptions.swift

```diff
@@ -3,12 +3,44 @@ import JavaScriptCore
 
 public struct ValidationOptions {
 
-  let disallowedFieldNames: Set<String>
+  struct DisallowedFieldNames {
+    let scalar: Set<String>
```

Comment:
This should be `allFields` not `scalar`. It does (and should) also apply to entity fields, so calling it `scalar` is misleading.

Reply:
Fair point, I'll change that up now - thanks.